### PR TITLE
add coding page and modals

### PR DESCRIPTION
**Pull request by mcuozzo19**

### CODE REVIEW

File: portfolio/src/main/webapp/gallery.html [hunk 1]
```diff
@@ -21,19 +22,19 @@
 <nav class="navbar navbar-inverse">
   <div class="container-fluid">
     <div class="navbar-header">
-      <a class="navbar-brand" href="#">Madeline Cuozzo</a>
+      <a class="navbar-brand" href="#" style="font-family: Lekton">Madeline Cuozzo</a>
```
Comment (sapphire007-corp) [inline portfolio/src/main/webapp/gallery.html:25]: Update the style sheet to add font to an existing class. Can't quite tell if you want the same font all over or not, you could put it in higher level container class, if that is so.

File: portfolio/src/main/webapp/coding.html [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    and Udemy's Intro to Javascript and JS React.">
+    <img class="myImages" id="myImg" src="../images/Internships.png" id="IntImg" alt="I am currently a STEP intern at Google!">
+  </div>
+  <div id="myModal" class="modal">
+  <span class="close">&times;</span>
+  <img class="modal-content" id="img01">
+  <div id="caption"></div>
+</div>
+<script>
+    // create references to the modal...
```
Comment (sapphire007-corp) [inline portfolio/src/main/webapp/coding.html:66]: Fix indent.

File: portfolio/src/main/webapp/gallery.html [hunk 3]
```diff
@@ -58,7 +59,7 @@ <h1><b>My Life in Pictures</b></h1>
     <img src="../images/Soccer.jpeg" style="width:100%">
     <img src="../images/Hangingrock.JPG" style="width:100%">
     <img src="../images/Lobster1.JPG" style="width:100%">
-    <img src="../images/Mbs.JPG" style="width:100%">
+
```
Comment (sapphire007-corp) [inline portfolio/src/main/webapp/gallery.html:62]: Nit: remove the empty line

File: portfolio/src/main/webapp/style.css [hunk 4]
```diff
@@ -12,3 +12,127 @@
...
+  color: #bbb;
+  text-decoration: none;
+  cursor: pointer;
+}
+
+@media only screen and (max-width: 700px) {
+  .modal-content {
+    width: 100%;
+  }
+}
```
Comment (sapphire007-corp) [inline portfolio/src/main/webapp/style.css:138]: Nit: newline

File: portfolio/src/main/webapp/coding.html [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+<head>
+  
+  <title>Portfolio</title>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="stylesheet" type="text/css" href="//fonts.googleapis.com/css?family=Lekton" />
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
+  <link rel ="stylesheet" href = "style.css">
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
```
Comment (allyons-google) [inline portfolio/src/main/webapp/coding.html:12]: What are the licensing rules for using external scripts? If third party scripts isn't specifically advised by the STEP intern team, then you should probably create your own scripts for a few reasons.

File: portfolio/src/main/webapp/coding.html [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js"></script>
+  </head>
+<body>
+<nav class="navbar navbar-inverse">
+  <div class="container-fluid">
+    <div class="navbar-header">
+      <a class="navbar-brand" href="#" style="font-family: Lekton">Madeline Cuozzo</a>
```
Comment (allyons-google) [inline portfolio/src/main/webapp/coding.html:19]: Let's extract style rules into a separate CSS file.

File: portfolio/src/main/webapp/coding.html [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+      <li><a href="#" style="font-family: Lekton">Contact Me</a></li>
+    </ul>
+  </div>
+</nav>
+
+<h1 style="font-family: Lekton; font-style: bold">Coding</h1>
+
+<div class="row">
+  <div class="column">
+    <img class="myImages" id="myImg" src="../images/Langauges.png" id="langImg" alt=
```
Comment (allyons-google) [inline portfolio/src/main/webapp/coding.html:40]: IDs are generally not advisable, especially for reused elements. Classes are recommended for reusability.

File: portfolio/src/main/webapp/coding.html [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+            <img class="myImages" id="myImg" src="../images/Internships.png" id="IntImg" alt="I am currently a STEP intern at Google!">
+        </div>
+        <div id="myModal" class="modal">
+        <span class="close">&times;</span>
+        <img class="modal-content" id="img01">
+        <div id="caption"></div>
+        </div>
+    <script>
+    // create references to the modal...
+    var modal = document.getElementById('myModal');
```
Comment (allyons-google) [inline portfolio/src/main/webapp/coding.html:63]: Let's put scripts into a separate script file

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 9]
```diff
@@ -14,19 +14,75 @@
...
+    
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<String> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(maxComments))) {
+      
+      String text = (String) entity.getProperty("text");
```
Comment (allyons-google) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:57]: Defensive programming suggestion -- we should always check type casts. (In our production code, we additionally log messages to notify us if types are not what we expect.)

Comment (mcuozzo19) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:57]: Do you mean log to console? Also, what is the best way to check for type?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 10]
```diff
@@ -14,19 +14,75 @@
...
+      
+      comments.add(text);
+    }
+
+    Gson gson = new Gson();
```
Comment (allyons-google) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:62]: Do we need to create a new Gson instance on every get request?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 11]
```diff
@@ -14,19 +14,75 @@
...
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+     String maxCommentsString = request.getParameter("maxComments");
+
+    // Convert the input to an int.
+    int maxComments = Integer.parseInt(maxCommentsString);;
+    
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
```
Comment (allyons-google) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:49]: Can this be extracted from this method into a static or immutable variable?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 12]
```diff
@@ -14,19 +14,75 @@
...
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get the input from the form.
+    String newComment = request.getParameter("my-comment");
+    long timestamp = System.currentTimeMillis();
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", newComment);
+    commentEntity.setProperty("timestamp", timestamp);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (allyons-google) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:80]: Same as above, can we make this static or immutable?

File: portfolio/src/main/webapp/coding.html (hunk 8 above)
Comment (allyons-google) [reply portfolio/src/main/webapp/coding.html:63]: Is there a reason to mix script with template here?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 11 above)
Comment (mcuozzo19) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:51]: By this do you mean create a new method to return this query and call it in this function?

Comment (allyons-google) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:51]: I mean that if the query object can be created just once, then we may not need to create a new one each time doGet is called.

**Pull request merged**